components/service/glean/src/main/java/mozilla/components/service/glean/firstrun/FileFirstRunDetector.kt

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.glean.firstrun
+
+import android.support.annotation.VisibleForTesting
+import mozilla.components.support.base.log.logger.Logger
+import java.io.File
+import java.io.IOException
+
+/**
+ * A [FirstRunDetector] implementation that uses the presence
+ * of a specific file in the User's app directory to determine
+ * if this is a first run or not.
+ *
+ * @param gleanDataDir the directory, within the application's data dir, that
+ *        contains glean files.
+ */
+internal class FileFirstRunDetector(
+    private val gleanDataDir: String
```

Comment:
nit: Could be a `File`object. Would still allow you to do `val file = File(gleanDataDir, ..)`..

Reply:
Ah, nice!